pkg/controllers/cloud/account_poller.go

```diff
@@ -217,15 +227,50 @@ func (p *Poller) updateAccountPoller(name *types.NamespacedName, selector *cloud
 	return nil
 }
 
-func (p *accountPoller) doAccountPoller() {
+// RestartAccountPoller stops and starts a goroutine making sure there is only one account poller goroutine at a time.
+func (p *Poller) RestartAccountPoller(name *types.NamespacedName) error {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	accPoller, found := p.accPollers[*name]
+	if !found {
+		return fmt.Errorf("%s %v", errorMsgAccountPollerNotFound, name)
+	}
+
+	// Acquire mutex lock to make sure doAccountPolling is not running while new goroutine is started.
+	accPoller.mutex.Lock()
+	defer accPoller.mutex.Unlock()
```

Comment:
we need to release the lock at 247. If not, doAccountPolling will be blocked

Reply:
I had thought so and that is why initial code was like that. After you recommended to use defer, thought even it can work because in defer lock gets released and soon go-routine will be scheduled. 
PRobably its much safer to revert the code and release the lock soon after channel is closed.